server/src/main/java/org/apache/iotdb/db/qp/strategy/ExecuteSqlVisitor.java

```diff
@@ -0,0 +1,1271 @@
+              "LIMIT <N>: N must be a positive integer and can not be zero.");
+    }
+    if (ctx.offsetClause() != null) {
+      visit(ctx.offsetClause());
+    }
+    return initializedOperator;
+  }
+
+  @Override
+  public Object visitOffsetClause(TSParser.OffsetClauseContext ctx) {
+//    throw new SqlParseException("Not supported TSParser type KW_OFFSET");
+    return null;
+  }
+
+  @Override
+  public Object visitGroupbyClause(TSParser.GroupbyClauseContext ctx) {
+    selectOp = ((QueryOperator) initializedOperator).getSelectOperator();
+    if (selectOp.getSuffixPaths().size() != selectOp.getAggregations().size()) {
+      throw new SqlParseException(
+              "Group by must bind each seriesPath with an aggregation function");
+    }
+    ((QueryOperator) initializedOperator).setGroupBy(true);
+    long value = parseTimeUnit(ctx.value.getText(), ctx.unit.getText());
+    ((QueryOperator) initializedOperator).setUnit(value);
+
+    List<Pair<Long, Long>> intervals = new ArrayList<>();
+    for (TSParser.TimeIntervalContext interval : ctx.timeInterval()) {
+      intervals.add((Pair<Long, Long>) visit(interval));
+    }
+    ((QueryOperator) initializedOperator).setIntervals(intervals);
+
+    long originTime = 0;
+    if (ctx.timeOrigin == null) {
+      originTime = parseTimeFormat(SQLConstant.START_TIME_STR);
+    } else {
+      originTime = (long) visit(ctx.timeOrigin);
+    }
+    ((QueryOperator) initializedOperator).setOrigin(originTime);
+    return initializedOperator;
+  }
+
+  @Override
+  public Object visitTimeInterval(TSParser.TimeIntervalContext ctx) {
+    long startTime = (long) visit(ctx.dateFormatWithNumber(0));
+    long endTime = (long) visit(ctx.dateFormatWithNumber(1));
+    return new Pair<>(startTime, endTime);
+  }
+
+  @Override
+  public Long visitGetDateTimeFromDateFormat(TSParser.GetDateTimeFromDateFormatContext ctx) {
+    return parseTimeFormat((String) visit(ctx.dateFormat()));
+  }
+
+
+  @Override
+  public String visitGetDateTime(TSParser.GetDateTimeContext ctx) {
+    return ctx.DATETIME().getText();
+  }
+
+  @Override
+  public String visitGetIdentifier(TSParser.GetIdentifierContext ctx) {
+    return ctx.Identifier().getText();
+  }
+
+  @Override
+  public Object visitFillClause(TSParser.FillClauseContext ctx) {
+    fillTypes = new EnumMap<>(TSDataType.class);
+    for (TSParser.TypeClauseContext typeClauseContext : ctx.typeClause()) {
+      try {
+        fillClauseDataType = parseType(typeClauseContext.Identifier().getText());
+        visit(typeClauseContext.interTypeClause());
+      } catch (LogicalOperatorException e) {
+        throw new SqlParseException(e.getMessage());
+      }
+    }
+    ((QueryOperator) initializedOperator).setFillTypes(fillTypes);
+    ((QueryOperator) initializedOperator).setFill(true);
+    return initializedOperator;
+  }
+
+//  "select sensor1 from root.vehicle.device1 where time = 50 Fill(int32[linear, 5m, 5m], boolean[previous, 5m])"
```

Comment:
move this into the visitFillClause. Besides, if you have time, you can put an example in each visit() ...